source/SkiaSharp.Views.Uno/SkiaSharp.Views.Uno/SKSwapChainPanel.cs

```diff
@@ -0,0 +1,133 @@
+﻿using System;
+using Uno;
+using Windows.ApplicationModel;
+using Windows.Graphics.Display;
+using Windows.UI.Core;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Data;
+
+namespace SkiaSharp.Views.UWP
+{
+	public partial class SKSwapChainPanel : FrameworkElement
+	{
+		private static readonly DependencyProperty ProxyVisibilityProperty =
+			DependencyProperty.Register(
+				"ProxyVisibility",
+				typeof(Visibility),
+				typeof(SKSwapChainPanel),
+				new PropertyMetadata(Visibility.Visible, OnVisibilityChanged));
+
+		private static bool designMode = DesignMode.DesignModeEnabled;
+
+		private bool isVisible = true;
+
+		// workaround for https://github.com/mono/SkiaSharp/issues/1118
+		private int loadUnloadCounter = 0;
+
+		private void Initialize()
+		{
+			if (designMode)
+				return;
+
+			var display = DisplayInformation.GetForCurrentView();
+			OnDpiChanged(display);
+
+			Loaded += OnLoaded;
+			Unloaded += OnUnloaded;
+			SizeChanged += OnSizeChanged;
+
+			var binding = new Binding
+			{
+				Path = new PropertyPath(nameof(Visibility)),
+				Source = this
+			};
+			SetBinding(ProxyVisibilityProperty, binding);
+		}
+
+		public SKSize CanvasSize => GetCanvasSize();
+
+		public GRContext GRContext => GetGRContext();
+
+		public double ContentsScale { get; private set; }
+
+		[NotImplemented]
+		public bool DrawInBackground
+		{
+			get => throw new NotImplementedException();
+			set => throw new NotImplementedException();
+		}
+
+		[NotImplemented]
+		public bool EnableRenderLoop
+		{
+			get => throw new NotImplementedException();
+			set => throw new NotImplementedException();
+		}
+
+		public new void Invalidate()
+		{
+			if (Dispatcher.HasThreadAccess)
+				DoInvalidate();
+			else
+				_ = Dispatcher.RunAsync(CoreDispatcherPriority.Normal, DoInvalidate);
+		}
+
+		public event EventHandler<SKPaintGLSurfaceEventArgs> PaintSurface;
+
+		protected virtual void OnPaintSurface(SKPaintGLSurfaceEventArgs e)
+		{
+			// invoke the event
+			PaintSurface?.Invoke(this, e);
+		}
+
+		private static void OnVisibilityChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)
+		{
+			if (d is SKSwapChainPanel canvas && e.NewValue is Visibility visibility)
+			{
+				canvas.isVisible = visibility == Visibility.Visible;
+				canvas.Invalidate();
+			}
+		}
+
+		private void OnDpiChanged(DisplayInformation sender, object args = null)
+		{
+			ContentsScale = sender.LogicalDpi / 96.0f;
+			Invalidate();
+		}
+
+		private void OnSizeChanged(object sender, SizeChangedEventArgs e)
+		{
+			Invalidate();
+		}
+
+		private void OnLoaded(object sender, RoutedEventArgs e)
+		{
+			loadUnloadCounter++;
+			if (loadUnloadCounter != 1)
+				return;
```

Comment:
Not actually sure Uno also has the "feature-bug" https://github.com/mono/SkiaSharp/issues/1118